packages/common/src/getNonceManagerId.ts

```diff
@@ -10,6 +10,7 @@ export async function getNonceManagerId({
   address: Hex;
   blockTag: BlockTag;
 }): Promise<string> {
+  // TODO: improve this so we don't have to call getChainId every time
```

Comment:
right now, this is going to make an `eth_chainId` call per use of `writeContract` and `sendTransaction`

we can improve this by threading through a chain or chain ID from the upstream callers, but it's unclear what the "best practice" is in terms of viem client handling of chains (should they be mutable or not) and what to do for wallets that do have "mutable" chains (e.g. a Metamask wallet can switch network, but unclear if that should create  a new client object or if it should mutate the chain on the existing client object)

once I get clarity, I'll follow up and optimize this to remove the extra call